pkg/etcdctl/main.go

```diff
@@ -0,0 +1,85 @@
+package etcdctl
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/rancher/k3s/pkg/configfilearg"
+	"github.com/rancher/k3s/pkg/datadir"
+	"github.com/sirupsen/logrus"
+	"go.etcd.io/etcd/etcdctl/ctlv2"
+	"go.etcd.io/etcd/etcdctl/ctlv3"
+)
+
+const (
+	apiEnv = "ETCDCTL_API"
+	cacert = "ETCDCTL_CACERT"
+	cert   = "ETCDCTL_CERT"
+	key    = "ETCDCTL_KEY"
+)
+
+func Main() {
+	dataDir := findDataDir()
+	etcdCtlAPI := os.Getenv(apiEnv)
+	if etcdCtlAPI == "" {
+		os.Setenv(apiEnv, "3")
+	}
+	etcdCtlCaCert := os.Getenv(cacert)
+	if etcdCtlCaCert == "" {
+		os.Setenv(cacert, filepath.Join(dataDir, "server", "tls", "etcd", "server-ca.crt"))
+	}
+	etcdCtlCert := os.Getenv(cert)
+	if etcdCtlCert == "" {
+		os.Setenv(cert, filepath.Join(dataDir, "server", "tls", "etcd", "server-client.crt"))
+	}
+	etcdCtlKey := os.Getenv(key)
+	if etcdCtlKey == "" {
+		os.Setenv(key, filepath.Join(dataDir, "server", "tls", "etcd", "server-client.key"))
+	}
+	main()
+}
+
+func main() {
+	rand.Seed(time.Now().UnixNano())
+
+	apiv := os.Getenv(apiEnv)
+
+	// unset apiEnv to avoid side-effect for future env and flag parsing.
```

Comment:
Kubernetes has only supported etcd api v3 for years so I think we should just hardcode v3. We will never need to support v2.